LSF driver resource string can fail with host exclusion

If  rusage and select are already present in the LSF resource string, and a user supplies a list of hosts to exlude, the LSF driver will build an invalid string. 

This easily happens as both these are typically set through ert-configurations.

The existing parametrized test can be augmented to make this fail:
```
        pytest.param(
            "select[location=='cloud'] rusage[mem=7000]",
            ["rogue_host"],
            "select[location=='cloud' && hname!='rogue_host'] rusage[mem=7000]",
            id="select_and_rusage_in_resource_requirement_one_excluded_hosts",
        ),

```

```
    def test_build_resource_requirement_string(
        resource_requirement: Optional[str], exclude_hosts: List[str], expected_string: str
    ):
>       assert (
            build_resource_requirement_string(exclude_hosts, resource_requirement)
            == expected_string
        )
E       assert "select[locat...'rogue_host']" == 'select[locat...age[mem=7000]'
E         
E         - select[location=='cloud' && hname!='rogue_host'] rusage[mem=7000]
E         + select[location=='cloud'] rusage[mem=7000 && hname!='rogue_host']

tests/unit_tests/scheduler/test_lsf_driver.py:650: AssertionError

```

--- a/src/ert/scheduler/lsf_driver.py
+++ b/src/ert/scheduler/lsf_driver.py
@@ -130,33 +130,28 @@ def parse_bjobs(bjobs_output: str) -> Dict[str, JobState]:
 def build_resource_requirement_string(
     exclude_hosts: Sequence[str], resource_requirement: Optional[str]
 ) -> str:
-    exclude_hosts_string = ""
-    if exclude_hosts:
-        # Create a string representing the exclusion of hosts if any are provided.
-        select_list = [
-            f"hname!='{host_name}'" for host_name in exclude_hosts if host_name
-        ]
-        exclude_hosts_string = " && ".join(select_list)
+    exclude_clauses = (
+        [f"hname!='{host_name}'" for host_name in exclude_hosts if host_name]
+        if exclude_hosts
+        else []
+    )
 
-    # If no resource requirements are provided, simply return the exclusion string.
     if not resource_requirement:
-        return f"select[{exclude_hosts_string}]" if exclude_hosts_string else ""
+        return f"select[{' && '.join(exclude_clauses)}]" if exclude_clauses else ""
 
-    # If 'select[' is in the resource requirement string, insert the exclusion string.
-    if "select[" in resource_requirement and exclude_hosts_string:
-        # We split string into (before) "bla[..] bla[..] select[xxx_"
-        # and (after) "... bla[..] bla[..]". (we replaced one ']' with ' ')
-        # Then we make final string:  before + &&excludes] + after
-        end_index = resource_requirement.rindex("]")
-        first_part = resource_requirement[:end_index]
-        second_part = resource_requirement[end_index:]
-        return f"{first_part} && {exclude_hosts_string}{second_part}"
+    selects = re.match(r".*select\[(.*?)\].*", resource_requirement)
+    if selects and exclude_clauses:
+        select_clauses = selects[1].split("&&")
+        select_clauses = [string.strip() for string in select_clauses]
+        select_clauses.extend(exclude_clauses)
+        select_string = " && ".join(select_clauses)
+        return resource_requirement.replace(selects[1], select_string)
 
     # If 'select[' is not in the resource requirement, append the exclusion string.
-    if exclude_hosts_string:
-        return f"{resource_requirement} select[{exclude_hosts_string}]"
+    if exclude_clauses:
+        return f"{resource_requirement} select[{' && '.join(exclude_clauses)}]"
 
-    # Return the original resource requirement if no exclusions are needed.
+    # Return the original resource uequirement if no exclusions are needed.
     return resource_requirement
 
 

--- a/tests/unit_tests/scheduler/test_lsf_driver.py
+++ b/tests/unit_tests/scheduler/test_lsf_driver.py
@@ -610,7 +610,7 @@ async def test_that_bsub_will_retry_and_succeed(
             "select[location=='cloud']",
             ["linrgs12-foo", "linrgs13-bar"],
             "select[location=='cloud' && hname!='linrgs12-foo' && hname!='linrgs13-bar']",
-            id="multiple_selects",
+            id="existing_select",
         ),
         pytest.param(
             None,
@@ -629,6 +629,24 @@ async def test_that_bsub_will_retry_and_succeed(
             [""],
             "select[location=='cloud']",
             id="select_in_resource_requirement_and_empty_string_in_excluded_hosts",
+        ),
+        pytest.param(
+            "select[location=='cloud'] rusage[mem=7000]",
+            [""],
+            "select[location=='cloud'] rusage[mem=7000]",
+            id="select_and_rusage_in_resource_requirement_empty_excluded_hosts",
+        ),
+        pytest.param(
+            "select[location=='cloud'] rusage[mem=7000]",
+            ["rogue_host"],
+            "select[location=='cloud' && hname!='rogue_host'] rusage[mem=7000]",
+            id="select_and_rusage_in_resource_requirement_one_excluded_hosts",
+        ),
+        pytest.param(
+            "rusage[mem=7000] select[location=='cloud']",
+            ["rogue_host"],
+            "rusage[mem=7000] select[location=='cloud' && hname!='rogue_host']",
+            id="rusage_and_select_resource_requirement_one_excluded_hosts",
         ),
     ],
 )

